### PR TITLE
Implement link checking endpoint for manuals

### PR DESCRIPTION
**Pull request by tsteer**
- Implement service class for creating link check reports for manuals
- Copy govspeak link extractor from whitehall
- Add link checker api to services
- Add link check report controller

### CODE REVIEW

File: app/services/link_check_report/create_service.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      Manual.find(link_reportable_id, user)
+    end
+  end
+
+  def uris
+    Govspeak::LinkExtractor.new(link_reportable.body).links
+  end
+
+  def call_link_checker_api
+    Services.link_checker_api.create_batch(uris)
```
Comment (SebAshton) [inline app/services/link_check_report/create_service.rb]: Needs the `webhook_uri` and a `webmock_secret` too

Comment (lukemorton) [reply app/services/link_check_report/create_service.rb]: We'll cover this in a subsequent PR once there is a webhook_uri to point to.

File: app/services/link_check_report/create_service.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  end
+
+  def call
+    link_report = call_link_checker_api
+
+    report = LinkCheckReport.new(
+      batch_id: link_report.fetch(:batch_id),
+      completed_at: link_report.fetch(:completed_at),
+      status: link_report.fetch(:status),
+      link_reportable_type: link_reportable_type,
```
Comment (thomasleese) [inline app/services/link_check_report/create_service.rb]: Should this be using `manual_id` and `section_id`?

File: spec/services/link_check_report/create_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,124 @@
+require "spec_helper"
+
+RSpec.describe LinkCheckReport::CreateService do
+  let(:user) { FactoryGirl.create(:user) }
+
+  let(:link_check_report) { double(:link_check_report, valid?: nil, save!: nil) }
```
Comment (thomasleese) [inline spec/services/link_check_report/create_service_spec.rb]: Would be good to use a Factory Bot class here to make sure the fields definitely exist on the model.

**Pull request closed without merging**